python/objToJSON.c

```diff
@@ -579,7 +592,7 @@ void Object_beginTypeContext (JSOBJ _obj, JSONTypeContext *tc)
     return;
   }
   else
-  if (PyString_Check(obj))
+  if (PyString_Check(obj) && !PyObject_HasAttrString(obj, "__json__"))
```

Comment:
So far I did follow. But not why one would need it. What's the actual use-case for that? Furthermore why would only the string type have a `__json__` method but not the other types?